src/dx/presence/builders/set_presence_state.rs

```diff
@@ -173,9 +193,16 @@ where
     /// Build and call asynchronous request.
     pub async fn execute(self) -> Result<SetStateResult, PubNubError> {
         let request = self.request()?;
+
+        #[cfg(feature = "std")]
+        if !request.channels.is_empty() {
+            request.on_execute.clone()(request.channels.clone(), request.state.clone());
```

Comment:
Do we need to clone the PTR here? It seems like it is not moved out from this call.